File: projects/frontend/src/types/customer.ts
export interface Customer {
  id: string;
  customerName: string | null;
  companyName: string | null;
  email: string;
  mobilePhone: string | null;
  phone: string | null;
  city: string | null;
  postCode: string | null;
  country: string | null;
  createdDate: string;
}

export interface CustomerDetail {
  id: string;
  customerName: string | null;
  companyName: string | null;
  email: string;
  mobilePhone: string | null;
  phone: string | null;
  addressLine1: string | null;
  addressLine2: string | null;
  city: string | null;
  county: string | null;
  postCode: string | null;
  country: string | null;
  createdDate: string;
  orderCount: number;
}

export interface CreateCustomerRequest {
  customerName?: string;
  companyName?: string;
  email: string;
  mobilePhone?: string;
  phone?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  county?: string;
  postCode?: string;
  country?: string;
}

export interface UpdateCustomerRequest {
  customerName?: string | null;
  companyName?: string | null;
  email: string;
  mobilePhone?: string | null;
  phone?: string | null;
  addressLine1?: string | null;
  addressLine2?: string | null;
  city?: string | null;
  county?: string | null;
  postCode?: string | null;
  country?: string | null;
}

File: projects/backend/src/EbayDesign.Domain/Entities/Customer.cs
using EbayDesign.Domain.Common;

namespace EbayDesign.Domain.Entities;

public class Customer : BaseEntity
{
    public string? CustomerName { get; set; }
    public string? EbayUsername { get; set; }
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? AddressLine1 { get; set; }
    public string? AddressLine2 { get; set; }
    public string? City { get; set; }
    public string? County { get; set; }
    public string? PostCode { get; set; }
    public string Country { get; set; } = "United Kingdom";

    public ICollection<Order> Orders { get; set; } = new List<Order>();
}
